javalib/src/main/scala/java/lang/Long.scala

```diff
@@ -200,51 +200,48 @@ object Long {
   @inline def compareUnsigned(x: scala.Long, y: scala.Long): scala.Int =
     compare(x ^ scala.Long.MinValue, y ^ scala.Long.MinValue)
 
+  private def fail(s: String): Nothing =
+    throw new NumberFormatException(s"""For input string: "$s"""")
+
   def decode(nm: String): Long = {
+    if (nm == null)
+      throw new NumberFormatException("null")
     val length = nm.length()
-    if (length == 0) {
-      throw new NumberFormatException()
-    } else {
-      var i          = 0
-      var firstDigit = nm.charAt(i)
-      val negative   = firstDigit == '-'
+    if (length == 0) fail(nm)
 
-      if (negative) {
-        if (length == 1) {
-          throw new NumberFormatException(nm)
-        }
+    var i          = 0
+    var firstDigit = nm.charAt(i)
+    val negative   = firstDigit == '-'
+    val positive   = firstDigit == '+'
 
-        i += 1
-        firstDigit = nm.charAt(i)
-      }
+    if (negative || positive) {
+      if (length == 1) fail(nm)
+      i += 1
+      firstDigit = nm.charAt(i)
+    }
 
-      var base = 10
-      if (firstDigit == '0') {
-        i += 1
-        if (i == length) {
-          return valueOf(0L)
-        }
+    var base = 10
+    if (firstDigit == '0') {
+      i += 1
+      if (i == length) {
+        return valueOf(0L)
+      }
 
-        firstDigit = nm.charAt(i)
-        if (firstDigit == 'x' || firstDigit == 'X') {
-          if (i == length) {
-            throw new NumberFormatException(nm)
-          }
-          i += 1
```

Comment:
This was a small bug. If we check the length after increasing the counter, we are never going to output the exception because the length was just 2 (e.g. input `0x` or `0X`)